tasks/Task.cpp

```diff
@@ -2,19 +2,57 @@
 
 #include "Task.hpp"
 
+#include <gps_base/BaseTypes.hpp>
+#include <gps_base/rtcm3.hpp>
+
 using namespace gps_ntrip;
+using namespace std;
+
+constexpr int PACKETIZER_BUFFER_SIZE = 4096;
+
+struct RTCMPacketizer : public iodrivers_base::Driver {
+    RTCMPacketizer()
+        : iodrivers_base::Driver(PACKETIZER_BUFFER_SIZE) {}
+
+    int extractPacket(uint8_t const* buffer, size_t size) const override {
+        return gps_base::rtcm3::extractPacket(buffer, size);
+    }
+};
 
 Task::Task(std::string const& name)
-    : TaskBase(name)
+: TaskBase(name)
 {
+    mRTCMPacketizer.reset(new RTCMPacketizer);
+    mRTCMPacketizer->openURI("test://");
+    mRTCMPacketizerStream = dynamic_cast<iodrivers_base::TestStream*>(
+        mRTCMPacketizer->getMainStream()
+    );
+
+    mNTRIP.set_report_interval(10);
+    mNTRIP.OnReceived([this] (const char *buffer, int size) {
+        uint8_t forwardBuffer[4096];
+
+        mRTCMPacketizerStream->pushDataToDriver(vector<uint8_t>(buffer, buffer + size));
```

Comment:
We use iodrivers_base::Driver as an interface to re-packetize a raw non-packetized stream, so that we always send whole RTCM packets on the wire.

The test stream "acts as" a I/O stream from the perspective of iodrivers_base. So, we push data to the driver "as-if" we were a device, and then read the packetized version of it from `readPacket`.